src/green/token.rs

```diff
@@ -4,11 +4,17 @@ use crate::{green::SyntaxKind, SmolStr, TextSize};
 
 #[repr(align(2))] // NB: this is an at-least annotation
 #[derive(Debug, PartialEq, Eq, Hash)]
-struct GreenTokenData {
+pub(crate) struct GreenTokenData {
     kind: SyntaxKind,
     text: SmolStr,
 }
 
+impl GreenTokenData {
+    pub fn new(kind: SyntaxKind, text: SmolStr) -> GreenTokenData {
```

Comment:
```suggestion
    pub(crate) fn new(kind: SyntaxKind, text: SmolStr) -> GreenTokenData {
```

(there's an assist to swtich `pub` to `pub(crate)`